src/ukify/ukify.py

```diff
@@ -824,6 +824,20 @@ def make_uki(opts):
     if pcrpkey is None:
         if opts.pcr_public_keys and len(opts.pcr_public_keys) == 1:
             pcrpkey = opts.pcr_public_keys[0]
+            # If we are getting a certificate when using an engine, we need to convert it to public key format
+            if opts.signing_engine is not None:
+                from cryptography.hazmat.primitives import serialization
+                from cryptography.x509 import load_pem_x509_certificate
+
+                try:
+                    cert = load_pem_x509_certificate(pathlib.Path(pcrpkey).read_bytes())
```

Comment:
I think they `try..except` should only cover this line. `x509.load_pem_x509_certificate` raises ValueError if the input is not a certificate, so catching ValueError seems appropriate. But if we get past this line, we don't expect failure, so we shouldn't suppress exceptions there.

Reply:
Hmm, so we don't actually know whether pcrpkey is a file at all. Don't we risk trying to do something very unexpected here if we try to read from a non-file argument?

Reply:
fixed